source/onednngraph.js

```diff
@@ -0,0 +1,515 @@
+
+var onednngraph = onednngraph || {};
+var json = json || require('./json');
+var base = base || require('./base');
+
+onednngraph.ModelFactory = class {
+
+    match(context) {
+        const identifier = context.identifier;
+        const extension = identifier.split('.').pop().toLowerCase();
+        if (extension === 'json') {
+            const obj = context.open('json');
+            if (obj && obj.version && obj.graph) {
+                return 'onednngraph.json';
+            }
+        }
+        return undefined;
+    }
+
+    open(context, match) {
+        return onednngraph.Metadata.open(context).then((metadata) => {
+            const createModel = (metadata, symbol, version) => {
+                return new onednngraph.Model(metadata, symbol, version);
+            };
+            switch (match) {
+                case 'onednngraph.json': {
+                    let symbol = null;
+                    try {
+                        symbol = context.open('json');
+                    }
+                    catch (error) {
+                        const message = error && error.message ? error.message : error.toString();
+                        throw new onednngraph.Error("Failed to load symbol entry (" + message.replace(/\.$/, '') + ').');
```

Comment:
`context.open` should always succeed and `catch` is not needed?

Reply:
yes, changed.